front/src/applications/operationalStudies/hooks/useAutoUpdateProjection.ts

```diff
@@ -0,0 +1,50 @@
+import { useEffect } from 'react';
+
+import { useSelector } from 'react-redux';
+
+import type { InfraWithState } from 'common/api/osrdEditoastApi';
+import type { TrainScheduleWithDetails } from 'modules/trainschedule/components/Timetable/types';
+import {
+  updateSelectedTrainId,
+  updateTrainIdUsedForProjection,
+} from 'reducers/osrdsimulation/actions';
+import { getSelectedTrainId, getTrainIdUsedForProjection } from 'reducers/osrdsimulation/selectors';
+import { useAppDispatch } from 'store';
+
+const useAutoUpdateProjection = (
+  infra: InfraWithState,
+  trainIds: number[],
+  trainScheduleSummaries: TrainScheduleWithDetails[]
+) => {
+  const dispatch = useAppDispatch();
+  const currentProjection = useSelector(getTrainIdUsedForProjection);
+  const selectedTrainId = useSelector(getSelectedTrainId);
+
+  useEffect(() => {
+    if (infra.state === 'CACHED' && trainIds.length > 0) {
+      // if a selected train is given, we use it for the projection
+      if (selectedTrainId && !currentProjection && trainIds.includes(selectedTrainId)) {
+        dispatch(updateTrainIdUsedForProjection(selectedTrainId));
+        return;
+      }
+
+      // if there is already a projection and the projected train still exists, keep it
+      if (currentProjection && trainIds.includes(currentProjection)) {
+        if (!selectedTrainId) dispatch(updateSelectedTrainId(trainIds[0]));
+        return;
+      }
+
+      // by default, use the first valid train
+      const firstValidTrain = trainScheduleSummaries.find((train) => train.isValid);
+      if (firstValidTrain) {
+        dispatch(updateTrainIdUsedForProjection(firstValidTrain.id));
+        dispatch(updateSelectedTrainId(firstValidTrain.id));
+      }
+    } else {
+      if (selectedTrainId) dispatch(updateSelectedTrainId(undefined));
+      if (currentProjection) dispatch(updateTrainIdUsedForProjection(undefined));
+    }
+  }, [trainIds, infra && trainScheduleSummaries]);
```

Comment:
what is this `&&` meant to do in the dependency array?

Reply:
Thanks, well spotted !